engine/table/src/main/java/io/deephaven/engine/table/impl/locations/local/URIListKeyValuePartitionLayout.java

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.impl.locations.local;
+
+import io.deephaven.api.util.NameValidator;
+import io.deephaven.base.verify.Assert;
+import io.deephaven.base.verify.Require;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.table.impl.locations.TableDataException;
+import io.deephaven.engine.table.impl.locations.TableLocationKey;
+import io.deephaven.engine.table.impl.locations.impl.TableLocationKeyFinder;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+import static io.deephaven.engine.table.impl.locations.local.KeyValuePartitionLayout.buildLocationKeys;
+
+/**
+ * Extracts a Key-Value partitioned layout from a list of URIs.
+ */
+public class URIListKeyValuePartitionLayout<TLK extends TableLocationKey> implements TableLocationKeyFinder<TLK> {
```

Comment:
The idea was to have a separate key-value partition layout which can process list of URIs. And then ParquetKeyValuePartitionLayout to extend from this class.
The current KeyValuePartitionLayout uses files, so I was hoping I can move the file specific part from KeyValuePartitionLayout to a FileKeyValuePartitionLayout and then get URIListKeyValuePartitionLayout to extend from KeyValuePartitionLayout. But KeyValuePartitionLayout might be getting used in Enterprise, so can't change it that easily. Will work with Andy to understand how they use the KeyValuePartitionLayout.